farm-boosted-rewards-subscriber/src/service.rs

```diff
@@ -33,51 +34,73 @@ pub trait ServiceModule:
     + only_admin::OnlyAdminModule
 {
     #[endpoint(subtractPayment)]
-    fn subtract_payment_endpoint(
-        &self,
-        service_index: usize,
-        user_ids: MultiValueEncoded<AddressId>,
-    ) {
+    fn subtract_payment_endpoint(&self, user_ids: MultiValueEncoded<AddressId>) {
         self.require_caller_is_admin();
-        require!(
-            service_index == SubscriptionUserType::Normal as usize
-                || service_index == SubscriptionUserType::Premium as usize,
-            "Invalid service index"
-        );
-        let is_premium_service = service_index == SubscriptionUserType::Premium as usize;
+        let current_epoch = self.blockchain().get_block_epoch();
+        let payment_recurrency = EPOCHS_IN_WEEK;
+        let standard_service_index = SubscriptionUserType::Normal as usize;
+        let premium_service_index = SubscriptionUserType::Premium as usize;
```

Comment:
Bad practice to use enums as values. Put this into a method of `SubscriptionUserType`, like `get_service_index(&self)`

Reply:
I know we usually avoid coding like this, but here I wanted to make sure it's clear we can only have these 2 options. I've changed the enum to two different constants. For how to code looks like now, I don't think a new method would be of much help.